packages/x-tree-view/src/internals/utils/tree.ts

```diff
@@ -259,3 +260,14 @@ export const getAllNavigableItems = (
 
   return navigableItems;
 };
+
+/**
+ * Checks if the event target is in a descendant of this item.
+ * This can prevent from firing some logic on the ancestors on the interacted item when the event handler is on the root.
+ * @param {React.UIEvent} event The event to check
+ * @param {HTMLElement | null} itemRoot The root of the item to check if the event target is in its descendants
+ * @returns {boolean} Whether the event target is in a descendant of this item
+ */
+export const isEventTargetInDescendants = (event: React.UIEvent, itemRoot: HTMLElement | null) => {
```

Comment:
@noraleonte let me know if you need something like this instead

```suggestion
export const isTargetInDescendants = (target: HTMLElement, itemRoot: HTMLElement | null) => {
  return itemRoot !== target.closest('*[role="treeitem"]');
}  
```

Reply:
Yeah, I think it's more versatile this way 👌